src/hobbits-plugins/importerexporters/UsbDevice/usbdevice.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "importexportinterface.h"
+#include "parameterdelegate.h"
+#include <libusb-1.0/libusb.h>
+
+typedef struct UsbParams
+{
+     //the pointer to the libusb device selected
+    libusb_device *dev;
+    //the pointer to the device list generated
+    libusb_device **devs;
+    //the pointer to the configuration descriptor generated
+    libusb_config_descriptor *config;
+    // the pointer to the libusb context for the active libusb session
+    libusb_context *ctx;
+    //the handle of the device used for transfers and device interactions
+    libusb_device_handle *handle;
+    //the number of the device selected
+    int deviceNum;
+    //the number of the interface selected
+    int interfaceNum;
+    //the number of the alternate setting selected
+    int altSetNum;
+    //number of the endpoint selected
+    int endpointNum;
+    //the address of the endpoint selected
+    unsigned char endpoint;
+    //any possible error codes passed
+    int errorCode;
+}StructName;
```

Comment:
I think the idiomatic way to do this would actually be:
```
typedef struct UsbParams {
...
} UsbParams;
```

you also might consider putting this struct definition inside the class as private since it is internal to the functionality of this class. This would also include making `setupLibusb` and `exitLibusb` private.